targets/TARGET_Realtek/TARGET_AMEBA/RTWInterface.cpp

```diff
@@ -191,85 +179,106 @@ nsapi_error_t RTWInterface::connect()
             return NSAPI_ERROR_PARAMETER;
     }
 
+    if (_channel > 0 && _channel < 14) {
+        uint8_t pscan_config = PSCAN_ENABLE;
+        wifi_set_pscan_chan(&_channel, &pscan_config, 1);
+    }
+
     ret = wifi_connect(_ssid, sec, _pass, strlen(_ssid), strlen(_pass), 0, (void *)NULL);
     if (ret != RTW_SUCCESS) {
         printf("failed: %d\r\n", ret);
         return NSAPI_ERROR_NO_CONNECTION;
     }
 
-    wlan_emac_link_change(true);
-    return mbed_lwip_bringup(_dhcp,
-                             _ip_address[0] ? _ip_address : 0,
-                             _netmask[0] ? _netmask : 0,
-                             _gateway[0] ? _gateway : 0);
+    if (!rtw_interface) {
+        nsapi_error_t err = rtw_obn_stack.add_ethernet_interface(rtw_emac, true, &rtw_interface);
+        if (err != NSAPI_ERROR_OK) {
+            rtw_interface = NULL;
+            return err;
+        }
+    }
+
+#define LWIP 0x11991199
+#if MBED_CONF_NSAPI_DEFAULT_STACK == LWIP
+    rtw_emac.wlan_emac_link_change(true);
```

Comment:
rtw_emac.wlan_emac_link_change() calls can be removed from here and link status call can be made on emac power up.

Reply:
@mikaleppanen 
the add_ethernet_interface will run emac_if_init to set the callback and then power up. Logic here is correct. 
However, the netif->flags will be set by the following code. It is still cause the "connection error" from  bringup().

Reply:
The netif->flags setting before power up was a bug in the lwip. I made a pull request to fix that: https://github.com/ARMmbed/mbed-os/pull/6926. 

You can remove the callback from there and take my pull request into your test build. It should work now correctly and trigger the link up from power up.

Reply:
Noted. Thank you.

Reply:
Thanks. Will add rtw_emac.wlan_emac_link_change(true); after wifi_connect .